martin/src/utils/id_resolver.rs

```diff
@@ -57,63 +60,72 @@ impl IdResolver {
-                info = info.map_or(String::new(), |v| format!(" ({v})"))
-            );
-        }
-        new_name
-    }
-
-    #[must_use]
-    fn resolve_int(&self, name: &str, unique_name: String) -> String {
-        // Ensure name has no prohibited characters like spaces, commas, slashes, or non-unicode etc.
-        // Underscores, dashes, and dots are OK. All other characters will be replaced with dashes.
-        let mut name = name.replace(
-            |c: char| !c.is_ascii_alphanumeric() && c != '_' && c != '.' && c != '-',
-            "-",
-        );
+        let stanitised_name = Self::sanitise(name);
 
+        let is_reserved_name = self.reserved.contains(stanitised_name.as_str());
         let mut names = self.names.lock().expect("IdResolver panicked");
-        if !self.reserved.contains(name.as_str()) {
-            match names.entry(name) {
+        // simple case if names need not be renamed
+        if !is_reserved_name {
+            match names.entry(stanitised_name.clone()) {
                 Entry::Vacant(e) => {
-                    let id = e.key().clone();
+                    if stanitised_name != name {
+                        warn!(
+                            "Source `{name}`{info} was renamed to `{stanitised_name}` because the ID may only contain alpha-numeric characters or `._-`",
+                            info = info.map_or(String::new(), |v| format!(" ({v})"))
+                        );
+                    }
                     e.insert(unique_name);
-                    return id;
+                    return stanitised_name;
                 }
-                #[allow(clippy::assigning_clones)]
                 Entry::Occupied(e) => {
-                    name = e.key().clone();
                     if e.get() == &unique_name {
-                        return name;
+                        return stanitised_name;
                     }
                 }
             }
         }
-        // name already exists, try it with ".1", ".2", etc. until the value matches
-        // assume that reserved keywords never end in a "dot number", so don't check
+
+        // need to rename => try ".1", ".2", etc. until the value matches
+        // assumes that reserved keywords never end in a "dot number" => no special case for this
         let mut index: i32 = 1;
         let mut new_name = String::new();
         loop {
             new_name.clear();
-            write!(&mut new_name, "{name}.{index}").unwrap();
+            write!(&mut new_name, "{stanitised_name}.{index}").unwrap();
             index = index.checked_add(1).unwrap();
             match names.entry(new_name.clone()) {
+                // found new name
                 Entry::Vacant(e) => {
+                    if is_reserved_name {
+                        warn!(
+                            "`{name}` is a reserved keyword, and may not be used as a source. The source was renamed to `{new_name}`{info}",
```

Comment:
Imo, this is more consistent with the other warnings 

```suggestion
                            "Source `{name}`{info} was renamed to `{new_name}` because {name} is a reserved keyword which may not be used as a source-ID",
```